testing/citest/spinnaker_testing/gcs_pubsub_trigger_agent.py

```diff
@@ -121,8 +121,22 @@ def detail(self):
 
   @property
   def timed_out(self):
+    ping = datetime.datetime.utcnow()
+    diff = ping - self.__start
+    return diff > self.timeout
+
+  @property
+  def timeout(self):
+    return self.__timeout
+
+  @timeout.setter
+  def timeout(self, val):
+    self.__timeout = val
+
+  def __check_finished_ok(self):
     resp = self.__trigger_response
-    return resp and resp.http_code == 504
+    executions = json.JSONDecoder().decode(resp.output) # Executions list
```

Comment:
If resp is not ok, is there resp.output?
If not then checks need to be separate.